src/Utilities/tests/test_ModernStringUtils.cpp

```diff
@@ -31,4 +31,61 @@ TEST_CASE("ModernStringUtils_strToLower", "[utilities]")
         "\xab"
         "not_just_ascii");
 }
+
+TEST_CASE("ModernStringUtils_split", "[utilities]")
+{
+  using modernstrutil::split;
+  std::string test_line{"hi there 101, random line"};
+  auto tokens = split(test_line, " ");
+  CHECK(tokens[0].size() == 2);
+  CHECK(tokens[4].size() == 4);
+  CHECK(tokens[3] == "random");
+
+  std::string test_lines{R"(
+this is a multi
+line
+token test
+)"};
+  auto tokens_lines = split(test_lines, "\n");
+  CHECK(tokens_lines[0] == "this is a multi");
+  CHECK(tokens_lines[1] == "line");
+  CHECK(tokens_lines[2] == "token test");
+
+  std::string test_multidel{"this \t is a multidelimiter  \n   \n token test"};
+  auto tokens_multidel = split(test_multidel, "\t \n");
+  CHECK(tokens_multidel[0] == "this");
+  CHECK(tokens_multidel[1] == "is");
+  CHECK(tokens_multidel[4] == "token");
+}
+
+TEST_CASE("ModernStringUtils_string2Real", "[utilities]")
+{
+  std::string_view svalue{"101.52326626"};
+  double value = string2Real<double>(svalue);
+  CHECK(value == Approx(101.52326626));
 } // namespace qmcplusplus
+
+TEST_CASE("ModernStringUtils_strip", "[utilities]")
```

Comment:
This should also have some more edge case tests - empty string, string that is only whitespace.

Reply:
Added